base_tier_validation/models/tier_review.py

```diff
@@ -77,11 +78,26 @@ def _compute_reviewed_formated_date(self):
 
     @api.depends("definition_id.approve_sequence")
     def _compute_can_review(self):
+        reviews = self.filtered(lambda rev: rev.status in ["waiting", "pending"])
+        if reviews:
+            # get minimum sequence of all to prevent jumps
+            next_seq = min(reviews.mapped("sequence"))
+            for record in reviews:
+                # if approve by sequence, check sequence has been reached
+                if record.approve_sequence:
+                    if record.sequence == next_seq:
+                        record.status = "pending"
+                # if there is no approval sequence go directly to pending state
+                elif not record.approve_sequence:
+                    record.status = "pending"
+                if record.status == "pending":
+                    if record.definition_id.notify_on_pending:
+                        record._notify_pending_status(record)
```

Comment:
I'm uncertain whether it's advisable to update fields within a `computed function` if those fields are not explicitly defined as computed (field `status`).

Reply:
>NOTE: Please let me know your comments regarding my comment on computed fields not being defined as computed explicitly. Could this be causing cache inconsistency?

tbh, I'm a bit overasked by this.. 
Maybe it is not the ideal situation. (yet, I'm unaware of any alternative method to implement this).
Well, we've deployed this code succesfully in production. W/o any problems. 
Judging from the experience it is okay..
But happy to learn from the more experienced dev's.